deforestation_api/routers/deforestation.py

```diff
@@ -0,0 +1,67 @@
+from shapely import Point
+import geopandas as gpd
+import pandas as pd
+import json
+
+from fastapi import APIRouter
+
+from deforestation_api.dependencies.deforestationdata import (
+    BasinDataDep,
+    LossyearDataDep,
+)
+from deforestation_api.dependencies.queryparams import CoordinatesDep, DateRangeDep
+
+router = APIRouter(prefix="/deforestation", tags=["deforestation"])
+
+
+def spatial_filter(
+    df: gpd.GeoDataFrame, lat: float, lon: float, predicate: str = "within"
+) -> gpd.GeoDataFrame:
+    p = Point(lon, lat)
+    query_index = df.sindex.query(p, predicate=predicate)
+    return df.iloc[query_index]
+
+
+def add_treecover_loss_data(
+    feature, treecover_loss: pd.DataFrame, startyear: int, endyear: int
+) -> None:
+    basin_loss = treecover_loss[treecover_loss["id"] == feature["id"]]
+    basin_loss = basin_loss.loc[basin_loss["year"].between(startyear, endyear)]
+    total_loss = basin_loss["area"].sum()
+    relative_loss = basin_loss["relative_area"].sum()
+    loss_per_year = basin_loss.drop(columns="id").to_json(orient="records")
+    loss_per_year = json.loads(loss_per_year)
+    feature["properties"]["daterange_tot_treeloss"] = total_loss
+    feature["properties"]["daterange_rel_treeloss"] = relative_loss
+    feature["properties"]["treeloss_per_year"] = loss_per_year
+
+
+@router.get("/basin")
```

Comment:
We need documentation for all routes, all input types and all return types from the api.